deno_webgpu/01_webgpu.js

```diff
@@ -510,6 +513,58 @@ ObjectDefineProperty(GPUShaderModule, customInspect, {
 });
 const GPUShaderModulePrototype = GPUShaderModule.prototype;
 
+ObjectDefineProperty(GPUCompilationInfo, customInspect, {
+  __proto__: null,
+  value(inspect, inspectOptions) {
+    return inspect(
+      createFilteredInspectProxy({
+        object: this,
+        evaluate: ObjectPrototypeIsPrototypeOf(
+          GPUCompilationInfoPrototype,
+          this,
+        ),
+        keys: [
+          "messages",
+        ],
+      }),
+      inspectOptions,
+    );
+  },
+});
+const GPUCompilationInfoPrototype = GPUCompilationInfo.prototype;
+
+ObjectDefineProperty(GPUCompilationMessage, customInspect, {
+  __proto__: null,
+  value(inspect, inspectOptions) {
+    return inspect(
+      createFilteredInspectProxy({
+        object: this,
+        evaluate: ObjectPrototypeIsPrototypeOf(
+          GPUCompilationMessagePrototype,
+          this,
+        ),
+        keys: [
+          "message",
+          "type",
+          "line_num",
+          "line_pos",
+          "offset",
+          "length",
+        ],
+      }),
+      inspectOptions,
+    );
+  },
+});
+const GPUCompilationMessagePrototype = GPUCompilationMessage.prototype;
+// Naming it `r#type` in Rust does not seem to work.
+// There is likely a problem with `GPUQuerySet.type` as well.
```

Comment:
issue: This should definitely get promoted to an issue, and not stay as a speculative comment.